src/diffusers/loaders.py

```diff
@@ -1201,12 +1218,8 @@ def _map_sgm_blocks_to_diffusers(cls, state_dict, unet_config, delimiter="_", bl
                 )
                 new_state_dict[new_key] = state_dict.pop(key)
 
-        if is_all_unet and len(state_dict) > 0:
+        if len(state_dict) > 0:
```

Comment:
Refactor here a bit by moving "adding text encoder keys" further up to have stronger check/assert here